sys/tiny_strerror/tiny_strerror.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *               2022 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Tiny strerror() implementation
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include "kernel_defines.h"
+#include "tiny_strerror.h"
+
+static const char *lookup[] = {
+    [0]                 = "OK",
```

Comment:
I wonder if a switch case would be more memory efficient here since the array will have gaps.

Reply:
There are a few gaps on newlib, as the few uncommon / non-standard errno codes are in between of the ones supported here. But I am pretty sure that the compiler will also create a look up table anyway. I have seen GCC allocating 512 bytes for a look up table of pointers to implement a `switch ()` statement on AVR in RAM despite `-Os` back then when [`saul_class_to_str()`](https://github.com/RIOT-OS/RIOT/blob/b2bcfabcfba3bc664e616817553a74ce40685e5f/drivers/saul/saul_str.c#L69-L99) was still a single large `switch ()` statement.

But now I wonder if we should provide our own `errno.h` that is optimized to be gap-free for the standard errno codes. I think for AVR we provide our own anyway.

Reply:
I guess we *really* need #18148 here then

Reply:
Sadly, we hardly can return `__flash const char *` from `strerror()`. But a single in-memory buffer would be a quick and dirty fix, that will badly break on concurrent use of `strerror()`. A `perror()` implementation would however be very much possible. Ideally, a variant that takes the error to print as argument rather than via the `errno` variable.